### PR TITLE
Various improvements

### PR DESCRIPTION
**Pull request by DemiMarie**
See individual commit messages for details.

### CODE REVIEW

**Comment by marmarek** [comment]
Mypy (I think rightfully) complains.

**Comment by DemiMarie** [comment]
PipelineRetryFailed

File: splitgpg2/__init__.py [hunk 1]
```diff
@@ -749,39 +766,40 @@ def update_keygrip_map(self) -> None:
...
+            if fields[0] in [b"sec", b""] and primary_key is not None:
+                keys.append(primary_key)
             if fields[0] == b"sec":
-                key = KeyInfo()
+                primary_key = KeyInfo(fields[11])
             elif fields[0] == b"ssb":
-                subkey = SubKeyInfo()
+                assert primary_key is not None, 'subkey before primary key?'
+                subkey = SubKeyInfo(fields[11], primary_key)
+                primary_key.subkeys.append(subkey)
```
Comment (marmarek) [inline splitgpg2/__init__.py:786]: Aren't you appending it twice now? It's already done above when finishing processing given subkey.

File: splitgpg2/test_server.py [hunk 2]
```diff
@@ -28,6 +28,9 @@
 from unittest import mock
 from . import GpgServer
 
+if not __debug__:
+    # Disabling assertions makes no sense when running tests
+    raise AssertionError('Assertions disabled while running tests?')
```
Comment (marmarek) [inline splitgpg2/test_server.py:33]: Please don't do that in Python either.

**Review by marmarek — CHANGES_REQUESTED** [review]

**Review by marmarek — APPROVED** [review]

**Pull request merged**